lib/interface_quda.cpp

```diff
@@ -3099,85 +3094,85 @@ void callMultiSrcQuda(void **_hp_x, void **_hp_b, QudaInvertParam *param, // col
         milc_fatlink_param.x[d] *= split_key[d];
         if (is_asqtad) milc_longlink_param.x[d] *= split_key[d];
       }
-      gauge_param.X[d] *= split_key[d];
-      if (!is_staggered) gauge_param.ga_pad *= split_key[d];
     }
 
-    // Deal with clover field. For Multi source computatons, clover field construction is done
-    // exclusively on the GPU.
-    quda::CloverField collected_clover;
-    bool is_clover = param->dslash_type == QUDA_CLOVER_WILSON_DSLASH || param->dslash_type == QUDA_TWISTED_CLOVER_DSLASH
-      || param->dslash_type == QUDA_CLOVER_HASENBUSCH_TWIST_DSLASH;
-
-    if (is_clover) {
-      if (param->clover_coeff == 0.0 && param->clover_csw == 0.0)
-        errorQuda("called with neither clover term nor inverse and clover coefficient nor Csw not set");
-      if (gaugePrecise->Anisotropy() != 1.0) errorQuda("cannot compute anisotropic clover field");
-
-      if (h_clover || h_clovinv) {
-        CloverFieldParam clover_param(*param, X);
-        clover_param.create = QUDA_REFERENCE_FIELD_CREATE;
-        clover_param.setPrecision(param->clover_cpu_prec);
-        clover_param.inverse = h_clovinv ? true : false;
-        clover_param.clover = h_clover;
-        clover_param.cloverInv = h_clovinv;
-        clover_param.order = param->clover_order;
-        clover_param.location = param->clover_location;
-        clover_param.reconstruct = false;
-
-        // Adjust inv_param->clover_coeff: if a user has set kappa and Csw,
-        // populate inv_param->clover_coeff for them as the computeClover
-        // routines uses that value
-        param->clover_coeff = (param->clover_coeff == 0.0 ? param->kappa * param->clover_csw : param->clover_coeff);
-
-        CloverField input_clover(clover_param);
-
-        for (int d = 0; d < CommKey::n_dim; d++) { clover_param.x[d] *= split_key[d]; }
-        clover_param.create = QUDA_NULL_FIELD_CREATE;
-        collected_clover = CloverField(clover_param);
-
-        quda::split_field(collected_clover, {input_clover}, split_key); // Clover uses 4d even-odd preconditioning.
-      }
-    }
-
-    quda::GaugeField collected_gauge;
-    quda::GaugeField collected_milc_fatlink_field;
-    quda::GaugeField collected_milc_longlink_field;
+    quda::GaugeField *collected_gauge = nullptr;
+    quda::GaugeField *collected_milc_fatlink_field = nullptr;
+    quda::GaugeField *collected_milc_longlink_field = nullptr;
 
     if (!is_staggered) {
       gf_param.create = QUDA_NULL_FIELD_CREATE;
-      collected_gauge = quda::GaugeField(gf_param);
-      quda::split_field(collected_gauge, {in}, split_key);
+      collected_gauge = new quda::GaugeField(gf_param);
```

Comment:
I think we don't need to use pointers to the gauge fields anymore - instead we can just use the constructor themselves.

Reply:
I remember we have some policy in QUDA that we should not use raw pointers anymore so might make sense to avoid all the raw pointers in the changes here - should be straight forward to do so I believe.

Reply:
We discussed this during our debug sesh. The reason here was that I use some pointer-swapping tricks to back-up the original unsplit fields, to set up the split fields and to then restore the originals. It seemed to make sense to use pointers here. However, I can try and think more carefully and not use those. I could in principle drop the pointers I have into a `std::unique_ptr` but the way we hold `gaugePrecise` and `cloverPrecise` etc. is also not done with smart pointers either... So I think in our conversation we ended up with saying pointer use is OK here.